### ISSUE TITLE
TypeError: if no direction is specified, key_or_list must be an instance of list with gridfs and pymongo 4.0

### ISSUE THREAD

**Issue by thequilo** [mongomock/mongomock#730]
This issue came up while disscussing failing tests in the sacred package in IDSIA/sacred#852. The provided example works with pymongo <4.0 but fails with pymongo 4.0.

Contents of `test.py`:
```python
import mongomock

db = mongomock.MongoClient().db
from mongomock.gridfs import enable_gridfs_integration

enable_gridfs_integration()

import gridfs

s = gridfs.GridFS(db)
s.put(b'hello world')
```
Error message:
```
$ python test.py
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    s.put(b'hello world')
  File ".../python3.8/site-packages/gridfs/__init__.py", line 135, in put
    grid_file.close()
  File ".../python3.8/site-packages/gridfs/grid_file.py", line 320, in close
    self.__flush()
  File ".../python3.8/site-packages/gridfs/grid_file.py", line 299, in __flush
    self.__flush_buffer()
  File ".../python3.8/site-packages/gridfs/grid_file.py", line 291, in __flush_buffer
    self.__flush_data(self._buffer.getvalue())
  File ".../python3.8/site-packages/gridfs/grid_file.py", line 272, in __flush_data
    self.__ensure_indexes()
  File ".../python3.8/site-packages/gridfs/grid_file.py", line 215, in __ensure_indexes
    self.__create_index(self._coll.files, _F_INDEX, False)
  File ".../python3.8/site-packages/gridfs/grid_file.py", line 209, in __create_index
    collection.create_index(
  File ".../python3.8/site-packages/mongomock/collection.py", line 1467, in create_index
    index_list = helpers.create_index_list(key_or_list)
  File ".../python3.8/site-packages/mongomock/helpers.py", line 84, in create_index_list
    raise TypeError('if no direction is specified, '
TypeError: if no direction is specified, key_or_list must be an instance of list
```

**Comment by oussjarrousse** [comment]
I have a good fix for this issue... But I am not sure how to contribute to this project...

**Comment by oussjarrousse** [comment]
I hope this helps @thequilo !

**Comment by oussjarrousse** [comment]
The issue here is that at the end of the `s.put()` command, the grid_fs client `s` is trying to ensure the collections `db.fs` and `db.chunck` has indexes, by calling `GridIn.__ensure_indexes()` in https://github.com/mongodb/mongo-python-driver/blob/master/gridfs/grid_file.py that in turns calls the GridIn`__create_index(self._coll.files, index_key, false)` with the `index_key = SON([("filename", ASCENDING), ("uploadDate", ASCENDING)])` as the keys parameter that eventually calls:                 `collection.create_index(index_key.items(), unique=unique, session=self._session)` that eventually causes the exception

note that the index_key.items() will return a dict.Items dictview object, not a list or a tuple or a string.

while pymongo.collection.create_index() will deal with this correctly, mongomock.collection.create_index will raise the exception mentioned above.

The fix I proposed in my pull request #731 fixes the issue by adding 2 lines of code from https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/helpers.py to https://github.com/mongomock/mongomock/blob/develop/mongomock/helpers.py

**Comment by thequilo** [comment]
@oussjarrousse Thank you! #731 resolves the `TypeError`.

**Comment by pcorpet** [comment]
Fixed in commit 151e12b
